lib/jekyll-admin/server/theme.rb

```diff
@@ -0,0 +1,117 @@
+module JekyllAdmin
+  class Server < Sinatra::Base
+    namespace "/theme" do
+      get do
+        ensure_theme
+        json(
+          {
+            :name        => theme.name,
+            :version     => theme.version,
+            :authors     => theme.authors.join(", "),
+            :license     => theme.license,
+            :path        => at_root,
+            :directories => entries.map(&:to_api),
+          }
+        )
+      end
+
+      get "/*?/?:path.:ext" do
+        ensure_theme
+        json(
+          { :name => filename,
+            :path => file_path, }.merge!(contents).merge!(url_fields)
+        )
+      end
+
+      get "/?*" do
+        ensure_theme && ensure_directory
+        json({
+          :name    => splats.first.split("/").last,
+          :path    => at_root(splats.first),
+          :entries => entries.map(&:to_api).concat(subdir_entries),
+        })
+      end
+
+      private
+
+      def ensure_theme
+        render_404 unless site.theme
+      end
+
+      def at_root(dir = "")
+        site.in_theme_dir(dir)
+      end
+
+      def theme
+        site.theme.send(:gemspec)
```

Comment:
I'm a bit hesitant to use a private API here. Why can't we just use the public API (which exposes name, version, etc.)?